### PR TITLE
Making prometheus remote config retries configurable

### PR DESCRIPTION
**Pull request by muaazsaleem**
Can be modified per cluster. Keeping the global config items to the
upstream defaults.

Signed-off-by: Muaaz Saleem <muhammad.muaaz.saleem@zalando.de>

### CODE REVIEW

**Comment by muaazsaleem** [comment]
👍

File: cluster/config-defaults.yaml [hunk 1]
```diff
@@ -180,10 +180,19 @@ prometheus_cpu: "1000m"
...
 
+# The remote configs match the prometheus defaults:
+# https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write
+prometheus_remote_write: "disabled"
+# Maximum time a sample will wait in buffer.
+prometheus_remote_batch_send_deadline: "5s"
+# Initial retry delay. Gets doubled for every retry.
+prometheus_remote_min_backoff: "30ms"
+# Maximum retry delay.
+prometheus_remote_max_backoff: "100ms"
```
Comment (szuecs) [inline cluster/config-defaults.yaml]: I would do 10s

File: cluster/config-defaults.yaml [hunk 2]
```diff
@@ -180,10 +180,19 @@ prometheus_cpu: "1000m"
...
 prometheus_tsdb_retention_size: "disabled"
 prometheus_csi_ebs: "false"
 
+# The remote configs match the prometheus defaults:
+# https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write
+prometheus_remote_write: "disabled"
+# Maximum time a sample will wait in buffer.
+prometheus_remote_batch_send_deadline: "5s"
+# Initial retry delay. Gets doubled for every retry.
+prometheus_remote_min_backoff: "30ms"
```
Comment (szuecs) [inline cluster/config-defaults.yaml]: I would do 3s

File: cluster/config-defaults.yaml [hunk 3]
```diff
@@ -180,10 +180,19 @@ prometheus_cpu: "1000m"
...
 prometheus_cpu_min: "0"
-prometheus_remote_write: "disabled"
 prometheus_tsdb_retention_size: "disabled"
 prometheus_csi_ebs: "false"
 
+# The remote configs match the prometheus defaults:
+# https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write
+prometheus_remote_write: "disabled"
+# Maximum time a sample will wait in buffer.
+prometheus_remote_batch_send_deadline: "5s"
```
Comment (szuecs) [inline cluster/config-defaults.yaml]: I would do 30s

**Comment by muaazsaleem** [comment]
👍

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**